### ISSUE TITLE
Array types in schemas displaying inconsistently

### ISSUE THREAD

**Issue by MatMoore** [zxchris/swaggerly#11]
This is a follow up to https://github.com/zxchris/swaggerly/issues/7 which was a broader question about resource pages. Since then I've created a few branches to experiment with different schemas.

One of the things I looked at was the handling of array types within the example spec, as I suspected I was using these wrong.

https://github.com/MatMoore/swaggerly/commit/7764d69c8efbeed40f923b80cf3a443c8cc70ae0
In this commit, I took the Activities resource from the existing example.

This currently renders an item named `history.activity-resource` with type `array`. I wondered if this is supposed to render as something like `history` / `array[activity]` instead.

I also added a "children" object to this resource with the same type (Activites) just to see what would swaggerly would do with it. This one rendered as `children` / `array`.

With other schemas I've seen this render differently, e.g. having type `array[array]` - see issue #13.

Ideally I'd prefer to render a property list table without the nested fields. I think there will be cases where inlining the child object fields adds too much noise, particularly if there are a lot of them. An alternative would be to display the type as something like `array[activity]` and just allow the user to click through to the type parameter.

**Comment by zxchris** [comment]
Hmm. The rendering of this model is definitely b0rked!
The flexibility of Swagger/OpenAPI gives you multiple ways to declare the same thing, which can make parsing complicated ;-) 

It will also be an interesting challenge to render recursive models, as you've played with in your test commit. Off the top of my head, a recursion check could be made and the rendered documentation made to explicitly refer to the nested type as being itself and going no further... something like that?

Back to looking at the array issue.....

**Comment by zxchris** [comment]
Fixed in branch merge/issues-and-multispec
Will merge into master shortly once documentation is complete.
